jumpscale/sals/reservation_chatflow/__init__.py

```diff
@@ -3,28 +3,51 @@
 from .solutions import solutions
 from jumpscale.sals.chatflows.chatflows import StopChatFlow
 from contextlib import ContextDecorator
+from jumpscale.sals.zos.zos import Zosv2
+from jumpscale.clients.explorer.models import WorkloadType
+from jumpscale.loader import j
+
+
+NODE_BLOCKING_WORKLOAD_TYPES = [
+    WorkloadType.Container,
+    WorkloadType.Network_resource,
+    WorkloadType.Volume,
+    WorkloadType.Zdb,
+]
 
 
 class DeploymentFailed(StopChatFlow):
-    def __init__(self, msg=None, solution_uuid=None, **kwargs):
+    def __init__(self, msg=None, solution_uuid=None, wid=None, **kwargs):
         super().__init__(msg, **kwargs)
         self.solution_uuid = solution_uuid
+        self.wid = wid
 
 
 class deployment_context(ContextDecorator):
     def __enter__(self):
         return self
 
     def __exit__(self, exc_type, exc, exc_tb):
-        if exc_type == DeploymentFailed and exc.solution_uuid:
+        if exc_type != DeploymentFailed:
+            return
+        if exc.solution_uuid:
+            # cancel related workloads
+            j.logger.info(f"canceling workload ids of solution_uuid: {exc.solution_uuid}")
             solutions.cancel_solution_by_uuid(exc.solution_uuid)
+        if exc.wid:
+            # block the failed node if the workload is network or container
+            zos = Zosv2()
```

Comment:
why we create a new instance from Zosv2 ? can we just 

`from jumpscale.sals.zos.zos import Zosv2 as zos` ?

Reply:
yes I need to create an object or I won't be able to access its properties